### ISSUE TITLE
Early bound generator adding enum property for Option set in Entity class

### ISSUE THREAD

**Issue by jagbiraquarian7** [daryllabar/DLaB.Xrm.XrmToolBoxTools#244]
**Tool and Version**
EarlyBoundGenerator V 1.2020.4.24

**Describe the bug**
after updating the version of the early bound generator we are unable to build the XRM project successfully. We are generating the entity and optionset classes from Dynamics CRM and using those classes to build our project.

After the update its generating a enum property for option set in the Entity class because of which when we build the project, error comes up : the namespace already contains a defination for the property.
==================================================
Below is the code of entity class  after update:
namespace APL.PigPass.XRM
{
	
	[System.Runtime.Serialization.DataContractAttribute()]
	[System.CodeDom.Compiler.GeneratedCodeAttribute("CrmSvcUtil", "9.1.0.42")]
	public enum UserQueryState
	{
		
		[System.Runtime.Serialization.EnumMemberAttribute()]
		Active = 0,
		
		[System.Runtime.Serialization.EnumMemberAttribute()]
		Inactive = 1,
	}
	
	/// <summary>
	/// Saved database query that is owned by a user.
	/// </summary>
	[System.Runtime.Serialization.DataContractAttribute()]
	[Microsoft.Xrm.Sdk.Client.EntityLogicalNameAttribute("userquery")]
	[System.CodeDom.Compiler.GeneratedCodeAttribute("CrmSvcUtil", "9.1.0.42")]
	public partial class UserQuery : Microsoft.Xrm.Sdk.Entity, System.ComponentModel.INotifyPropertyChanging, System.ComponentModel.INotifyPropertyChanged
	{
=================================================
Below is the Old code for the same entity class :
namespace APL.PigPass.XRM
{
	
	/// <summary>
	/// Saved database query that is owned by a user.
	/// </summary>
	[System.Runtime.Serialization.DataContractAttribute()]
	[Microsoft.Xrm.Sdk.Client.EntityLogicalNameAttribute("userquery")]
	[System.CodeDom.Compiler.GeneratedCodeAttribute("CrmSvcUtil", "9.0.0.9154")]
	[System.Data.Services.Common.DataServiceKeyAttribute("Id")]
	[System.Data.Services.IgnorePropertiesAttribute("Item", "Attributes", "EntityState", "FormattedValues", "RelatedEntities", "ExtensionData")]
	[Microsoft.Xrm.Client.Metadata.EntityAttribute("UserQuerySet")]
	public partial class UserQuery : Microsoft.Xrm.Client.CrmEntity
	{

===================================================
In the new code we can see and addition enum property for the Option set is added, which is causing the errors while building the project. It seems like the settings while generating the  classes have changed due to update and we are not able to map them correctly.

Below is the screenshot of the old settings:
![image](https://user-images.githubusercontent.com/65712571/82553388-42c0d980-9ba7-11ea-9a8b-71a8d93af943.png)

And below is the screenshot of new settings :
![image](https://user-images.githubusercontent.com/65712571/82553617-b82caa00-9ba7-11ea-93f0-7b1ac3b98666.png)

**To Reproduce**
Generating Entities and option sets using the latest EarlyBoundGenerator 

**Expected behavior**
Could you please help how can we generate the entity classes as before withouit having enum property for Option set.

**Screenshots**

**Additional context**

**Comment by daryllabar** [comment]
The generation of the state enum for the entity has always been done within the entity, since this is the way that the CRMSVCUTIL works by default.  Are you sure there wasn't anything that was done to remove these enums?  Has it always been just the Earlybound Generator being used?

**Comment by jagbiraquarian7** [comment]
Yes, we did not do anything to remove the enums. The only difference is the
settings have different options now may be that is something we are missing.
We have been using the early bound generator ever since the project
was built.

**Comment by daryllabar** [comment]
Ahh... I see what has happened, you were using the Xrm client which was removed in the CRM 7 SDK I believe.  I was having an issue with connecting to CRM with the CrmSvcUtil and was instructed by microsoft to remove that dependency.  So unfortunetly, to maintain compatibility with the newer versions of the CrmSvcUtil, I don't believe this is a change that I will be capable of undoing.

This gives you two options:
1. Remove your dependency on the Microsoft.Xrm.Client.  Delete all of your auto generated Entity and OptionSet files, use the latest and greatest version of the EBG to regenerate everything, and then fix all of your build errors in your code that has the old dependencies.
2.  Use an older version of the EarlyBoundGenerator.  I made the change on 3/30, so you'll have to manually download a version before 1.2020.3.30 from NuGet, extract it, and manually install that version into you XTB, and never update it.
